### PR TITLE
skip normal templating if is pdf or genpdf

### LINKED ISSUE #146: (SproutPRO) Can't access PDF's with Crio enabled
Can't view PDF's from an invoice or estimate with Crio
=========================================================
Name           |   PDF display
Reporter       |   Elana D.
Submit Date |   10/5/2022
Summary      | Can't view pdf's
Reproduction steps | install Crio, Install Sprout Invoices Pro (Business +)

- go into Sprout Invoices -> add-on's 
- Enable PDF Service for WordPress Invoices & Estimates
- Create a new Estimate
- Set a quanity, Rate
- Check Redirect to Invoice in the Estimnate Acceptance Actions
- View Estimate 
- Click on the PDF and verify it looks like the estimate
- Click Accept Estimate
- Should redirect you to the invoice and you should see the PDF icon
- Click on the PDF and verify it looks like the invoice 

The PDF's are saved in this directory and if you view the PDF that way it works but doing it from the estimate / invoice it doesn't. 
wp-content/uploads/si-pdfs/2022/10/estimate_304.pdf

### PR DESCRIPTION
**Pull request by jamesros161**
Resolves #146 

### Testing Instructions ###
1. install Crio test zip,
[crio.zip](https://github.com/BoldGrid/prime/files/9902903/crio.zip)

2. Install Sprout Invoices Pro (Business +) and activate license
3. go into Sprout Invoices -> add-on's
4. Enable PDF Service for WordPress Invoices & Estimates
5. Create a new Estimate
6. Set a quanity, Rate
7. Check Redirect to Invoice in the Estimnate Acceptance Actions
8. View Estimate
9. Click on the PDF and verify it looks like the estimate
10. Click Accept Estimate
11. Should redirect you to the invoice and you should see the PDF icon
12. Click on the PDF and verify it looks like the invoice

### CODE REVIEW

File: base.php [hunk 1]
```diff
@@ -114,3 +115,11 @@
 	</body>
 				<?php } ?>
 </html>
+<?php } elseif ( $is_sa_estimate ) {
```
Comment (bwmarkle) [inline base.php:118]: There's 100's of themes and plugins out there. Is the plan to continue adding additional `elseif` statements for each one?

File: base.php [hunk 2]
```diff
@@ -28,6 +28,7 @@
 $has_sticky_template = apply_filters( 'crio_premium_get_sticky_page_header', get_the_ID() );
 $has_sticky_template = get_the_ID() === $has_sticky_template ? false : $has_sticky_template;
 
+if ( ! isset( $_REQUEST[ 'pdf' ] ) && ! isset( $_REQUEST[ 'genpdf'] ) ) {
```
Comment (bwmarkle) [inline base.php:31]: Isn't this going to break any url that has `pdf=` or `genpdf=` parameters?

You're also stating that any site using Crio can't have a form submitted via post with a `pdf` value? What if someone is uploading a pdf to a site, and the form is using `<input type="file" name="pdf" id="pdf" ... />`

File: base.php [hunk 3]
```diff
@@ -114,3 +115,11 @@
...
 				<?php } ?>
 </html>
+<?php } elseif ( $is_sa_estimate ) {
+	$template = SI_Templating_API::override_template( 'estimate' );
+	load_template( $template );
+} elseif ( $is_sa_invoice ) {
+	$template = SI_Templating_API::override_template( 'invoice' );
+	load_template( $template );
+}
+?>
```
Comment (bwmarkle) [inline base.php:125]: Should we address the `no new line at end of file` warning?

**Review by bwmarkle — CHANGES_REQUESTED** [review]

File: base.php (hunk 1 above)
Comment (jamesros161) [reply base.php]: This logic has been moved out of the template file.

File: base.php (hunk 2 above)
Comment (jamesros161) [reply base.php]: The logic for this has been changed, and moved out of the base.php file.

**Review by avonville — DISMISSED** [review]

**Comment by nicolepaschen** [comment]
acceptance testing complete

**Pull request merged**